slick/src/main/scala/slick/basic/BasicBackend.scala

```diff
@@ -309,45 +309,47 @@ trait BasicBackend { self =>
           val debug = streamLogger.isDebugEnabled
           var state = initialState
           ctx.readSync
-          if(state eq null) acquireSession(ctx)
-          var demand = ctx.demandBatch
-          var realDemand = if(demand < 0) demand - Long.MinValue else demand
-          do {
-            try {
-              if(debug)
-                streamLogger.debug((if(state eq null) "Starting initial" else "Restarting ") + " streaming action, realDemand = " + str(realDemand))
-              if(ctx.cancelled) {
-                if(ctx.deferredError ne null) throw ctx.deferredError
-                if(state ne null) { // streaming cancelled before finishing
+          try {
+            if(state eq null) acquireSession(ctx)
```

Comment:
Here is where the bug occurs. `aquireSession` will throw an exception if the DB is down, but at the point in time, the `inUseCount` inside `ManagedArrayBlockingQueue` is already increased. 

Because we are not protecting against it, the `PrioritizedRunnable` won't get it's `connectionReleased` field set to `true` and as a consequence, the count will never be decreased.

We have observed it in a Lagom application that uses the `akka-persistence-jdbc` plugin. Because Lagom will constantly poll the event journal, after some time all threads (and connections) will be considered in-use. The `ManagedArrayBlockingQueue` will be permanently in paused stated because increase and decrease will never be balanced.

Reply:
For the record. Here is the issue in Lagom https://github.com/lagom/lagom/issues/1250